src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1141,6 +1141,8 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
             {
                 if (conn.TryGetBoundConnector(out var connector))
                 {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    // We cannot pass a token here, as we'll cancell a non-send query
```

Comment:
```suggestion
                    // We cannot pass a token here, as we'll cancel a non-send query
```

Reply:
I almost forgot about this invariant. Maybe we should have a common like:

// We don't pass the cancellation token to StartUserAction, since that would make it scope to the entire action (command execution), whereas it should only be scoped to the Execute method.

I'll also try to add some debug-only invariant checks to make sure we don't violate them